### PR TITLE
Update SASSAFRAS to SAFROLE

### LINKED ISSUE #5967: Update SASSAFRAS to SAFROLE
As it stands, SASSAFRAS will not be part of Polkadot protocol and the plan is to rollout a modified version SAFROLE thorugh JAM. Create a page for SAFROLE and redirect this page to it.

https://wiki.polkadot.network/docs/learn-sassafras

### PR DESCRIPTION
**Pull request by CrackTheCode016**
Closes #5967 

---

- Adds redirect from `/docs/learn-sassafras` to `/docs/learn-safrole`
- Updates mentions of SASSAFRAS 
- Moves explanation of SAFROLE from JAM page to the SAFROLE page, and links it in the JAM page

### CODE REVIEW

File: docs/learn/learn-jam-chain.md [hunk 1]
```diff
@@ -118,7 +118,7 @@ hash. This feature is the only opinionated aspect of the refine function.
 
 Tickets serve as anonymous entries into the block production mechanism. They are not immediately
 required for block production; instead, the system operates two epochs in advance. This mechanism is
-part of the SAFROL algorithm, a refined version of the original [SASSAFRAS](./learn-sassafras.md)
+part of the SAFROL algorithm, a refined version of the original [SASSAFRAS](./learn-safrole.md)
```
Comment (profradha) [inline docs/learn/learn-jam-chain.md]:
```suggestion
part of the SAFROLE algorithm, a refined version of the original [SASSAFRAS](./learn-safrole.md)
```

**Review by DrW3RK — COMMENTED** [review]
The content for SAFROLE can be sourced from https://graypaper.com/graypaper.pdf and it should be added as the resource link

File: docs/learn/learn-safrole.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    hybrid consensus,
+    block production,
+    sassafras,
+    safrole,
+    babe,
+  ]
+slug: ../learn-safrole
+---
+
+> This page will be updated as progress ensues.
```
Comment (filippoweb3) [inline docs/learn/learn-safrole.md]: I would remove this. Non-necessary info for the reader, and usually we make sure the Wiki is up-to-date (more or less) ;) There are other instances in the Wiki and will add this as part of the wiki reorg

File: docs/learn/learn-safrole.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+SAFROLE is a SNARK-based block production algorithm. It uses SNARK specifically for their anonymity
+features. And it delivers constant time block production, almost entirely fork-free. There are a
```
Comment (filippoweb3) [inline docs/learn/learn-safrole.md]: I would add the "why" here. Afaik anonymity is used to protect block producers' identity from being used to attack

**Review by filippoweb3 — CHANGES_REQUESTED** [review]
thank you Bader! Added a few comments

File: docs/learn/learn-safrole.md (hunk 3 above)
Comment (CrackTheCode016) [reply docs/learn/learn-safrole.md]: Added quite a few details, please see if it makes sense!

**Comment by CrackTheCode016** [comment]
@filippoweb3 I noticed that the Polkadot Direction Page still mentions SASSAFRAS. Not sure if we should update the mention, so I left it as-is (besides updating the link to reference `safrole.md` instead), let me know what you would like to do with it.

**Comment by filippoweb3** [comment]
@CrackTheCode016 yeah I would remove mention of sassafras in other pages as well. Thanks!

**Review by filippoweb3 — APPROVED** [review]

**Pull request merged**